src/coreclr/jit/codegenarm64.cpp

```diff
@@ -4760,13 +4760,35 @@ void CodeGen::genCodeForSelect(GenTreeOp* tree)
         prevCond = GenCondition::NE;
     }
 
+    regNumber               targetReg = tree->GetRegNum();
+    const GenConditionDesc& prevDesc  = GenConditionDesc::Get(prevCond);
+    regNumber               srcReg1   = op1->IsIntegralConst(0) ? REG_ZR : genConsumeReg(op1);
+
+    // When abs(op2 - op1) = 1, emit cinc instaed of csel. Lowering ensures op1 is the smaller constant.
+    if (op1->IsCnsIntOrI() && op2->IsCnsIntOrI() && op2->isContained() && opcond->OperIsSimple() &&
+        !varTypeIsFloating(opcond->gtGetOp1()->TypeGet()))
```

Comment:
@jakobbotsch - these checks are mostly just repeating what has already been checked in lowering (as can't just rely on contained being set). Do you think it's worth adding a `GT_CINC` LIR node?

Reply:
Cool, I'll update the PR accordingly.

Reply:
I was just thinking about https://github.com/dotnet/runtime/pull/82020 - and I released it'll affect this PR too.

With the current IR, for `CINC` we'd want a 2 op node which takes in a compare and a variable.

Post `82020` and the follow on `SELECTCC` additions, we'd then want both `CINC` and `CINCCC` nodes? `CINC` would be for the `SELECT` case and `CINCCC` would be for `SELECTCC` case

I guess it doesn't matter which order it's done in - this patch first or `GT_SELECTCC`, but whichever is the latter will have to add `CINCCC`.

Reply:
Either that or we represent `CINC` as `CMP(cond, 0)` + `CINCCC`. Or we stick with the containment approach from this PR. All those options are ok with me.

`GT_EQ` and co., and `GT_SELECT` would be representable this way too but it would not be compatible with some of the peepholes we do in the xarch backend.

Reply:
Using `CINC` and `CINCC` now. Marking as resolved!